omod/src/main/java/org/openmrs/module/appui/rest/SessionController.java

```diff
@@ -56,10 +57,12 @@ public Object get(HttpServletRequest request, HttpServletResponse response){
     public Object set(@RequestBody Map<String, Object> requestBody,
                       HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-        if(requestBody.get("location") != null){
+        if(requestBody.get("location") != null && StringUtils.isNotBlank(requestBody.get("location").toString())){
```

Comment:
Yes, but in this situation we first need to create these String doing this `requestBody.get("location").toString()` and during that the NPE is thrown